client/bgp_peer_test.go

```diff
@@ -20,6 +18,17 @@ var nextHopChoice string = "default"
 func TestAddBgpPeerAndDeleteBgpPeer(t *testing.T) {
 	c := NewClient(GetConfigFromEnv())
 
+	instanceName := "peer-test"
+	bgpPeerName := "test-peer"
+
+	_, err := c.AddBgpInstance(&BgpInstance{Name: instanceName, As: 65530, RouterID: "172.16.0.254"})
+	if err != nil {
+		t.Fatalf("unable to create BGP instance used for testing: %v", err)
+	}
+	defer func(c *Mikrotik, name string) {
```

Comment:
Have you verified that this works? I know that all deferred functions should run if this is called (reference [here](https://cs.opensource.google/go/go/+/refs/tags/go1.17:src/testing/testing.go;l=723;drc=refs%2Ftags%2Fgo1.17)), but I think the deferred function might need to come before calling `t.Fatalf`.